setup.py

```diff
@@ -60,7 +60,8 @@
 MICRO = 0
 ISRELEASED = False
 VERSION = '%d.%d.%d' % (MAJOR, MINOR, MICRO)
-REQUIRES = ['numpy (>=1.8)', 'scipy (>=0.15)', 'cython (>=0.21)']
+REQUIRES = ['numpy (>=1.8)', 'scipy (>=0.15)', 'cython (>=0.21)','matplotlib(>=1.2.1)']
```

Comment:
Does this line not make this a required, rather than optional module?

Reply:
It was also my question. I wonder whether the presence of INSTALL_REQUIRES and REQUIRES is redundant?

https://stackoverflow.com/questions/41268863/difference-between-extras-require-and-install-requires-in-setup-py 
There might be differences between `python setup.py install` and `pip install qutip` or `conda install qutip`.  
https://github.com/pypa/pip/issues/3147?

Reply:
I just tried looking up this `requires` argument. Cannot find much about. Seems maybe it is / was used by `distutils` rather than `setuptools`. I cannot remember why we still have the `distutils` option (see lines 34-47), most discussions say use `setuptools` over `distutils`.
@cgranade  was the last person to do an overhaul of our `setup.py`
I would suggest that this PR would be fine only adding the `extras_require`